Source/JavaScriptCore/runtime/StructureInlines.h

```diff
@@ -733,25 +740,32 @@ ALWAYS_INLINE Structure* Structure::addPropertyTransitionToExistingStructureConc
     return addPropertyTransitionToExistingStructureImpl(structure, uid, attributes, offset);
 }
 
-inline Structure* StructureTransitionTable::singleTransition() const
+inline Structure* StructureTransitionTable::trySingleTransition() const
 {
-    ASSERT(isUsingSingleSlot());
-    if (WeakImpl* impl = this->weakImpl()) {
-        if (impl->state() == WeakImpl::Live)
-            return jsCast<Structure*>(impl->jsValue().asCell());
-    }
+    uintptr_t pointer = m_data;
+    if (pointer & UsingSingleSlotFlag)
```

Comment:
This is intentional. This function can be called concurrently for GC marking. Thus, we need to avoid the  case, that m_data gets rewritten after using `isUsingSingleSlot()`.

Reply:
Ah, right.  I missed that this is caching `m_data` in `pointer`.